internals/overlord/state/notices.go

```diff
@@ -192,6 +228,13 @@ type AddNoticeOptions struct {
 	Time time.Time
 }
 
+func optionIntToString(maybeInt *int) string {
```

Comment:
Two nits:

1) Can we move this below, beside the other helper functions?
2) In Go you don't really have options/optionals/maybe, so how about the name `intPtrToString`? And `intPtr` for the parameter name.